File: packages/elixir-sdk/lib/tree_db_sdk/adapters/repositories.ex
defmodule TreeDbSdk.Repositories do
  @moduledoc false
  alias TreeDbSdk.Adapters.Common

  def register(client, body) do
    Common.json_request(client, :post, "/api/v1/repos/register", body, %{})
  end

  def list(client) do
    Common.json_request(client, :get, "/api/v1/repos", nil, %{})
  end

  def create(client, body) do
    Common.json_request(client, :post, "/api/v1/repos", body, %{})
  end

  def get(client, repo_id) do
    Common.json_request(client, :get, "/api/v1/repos/" <> Common.segment(repo_id), nil, %{})
  end

  def status(client, repo_id) do
    Common.json_request(
      client,
      :get,
      "/api/v1/repos/" <> Common.segment(repo_id) <> "/status",
      nil,
      %{}
    )
  end

  def refs(client, repo_id) do
    Common.json_request(
      client,
      :get,
      "/api/v1/repos/" <> Common.segment(repo_id) <> "/refs",
      nil,
      %{}
    )
  end

  def remotes(client, repo_id) do
    Common.json_request(
      client,
      :get,
      "/api/v1/repos/" <> Common.segment(repo_id) <> "/remotes",
      nil,
      %{}
    )
  end

  def push(client, repo_id, body \\ %{}),
    do:
      Common.json_request(
        client,
        :post,
        "/api/v1/repos/" <> Common.segment(repo_id) <> "/push",
        body
      )

  def sync(client, repo_id, body \\ %{}),
    do:
      Common.json_request(
        client,
        :post,
        "/api/v1/repos/" <> Common.segment(repo_id) <> "/sync",
        body
      )
end

File: packages/ts-sdk/src/treedx/adapters/federation/repositories.ts
import { jsonRequest, segment, type TreeDxAdapterContext } from '../runtime/common.js';

export class RepositoriesAdapter {
  constructor(private readonly context: TreeDxAdapterContext) {}
  register(input: unknown): Promise<unknown> { return jsonRequest(this.context.transport, 'POST', '/api/v1/repos/register', input); }
  list(): Promise<unknown> { return jsonRequest(this.context.transport, 'GET', '/api/v1/repos'); }
  create(input: unknown): Promise<unknown> { return jsonRequest(this.context.transport, 'POST', '/api/v1/repos', input); }
  get(repoId: string): Promise<unknown> { return jsonRequest(this.context.transport, 'GET', `/api/v1/repos/${segment(repoId)}`); }
  status(repoId: string): Promise<unknown> { return jsonRequest(this.context.transport, 'GET', `/api/v1/repos/${segment(repoId)}/status`); }
  refs(repoId: string): Promise<unknown> { return jsonRequest(this.context.transport, 'GET', `/api/v1/repos/${segment(repoId)}/refs`); }
  remotes(repoId: string): Promise<unknown> { return jsonRequest(this.context.transport, 'GET', `/api/v1/repos/${segment(repoId)}/remotes`); }
  push(repoId: string, input?: unknown): Promise<unknown> { return jsonRequest(this.context.transport, 'POST', `/api/v1/repos/${segment(repoId)}/push`, input); }
  sync(repoId: string, input?: unknown): Promise<unknown> { return jsonRequest(this.context.transport, 'POST', `/api/v1/repos/${segment(repoId)}/sync`, input); }
}
